src/source/PeerConnection/Rtp.c

```diff
@@ -159,25 +163,26 @@ STATUS writeFrame(PRtcRtpTransceiver pRtcRtpTransceiver, PFrame pFrame)
     UINT64 rtpTimestamp = 0;
     UINT64 now = GETTIME();
 
+    // stats updates
+    DOUBLE fps = 0.0;
+    UINT32 frames = 0, keyframes = 0, bytesSent = 0, packetsSent = 0, headerBytesSent = 0, framesSent = 0, hugeFramesSent = 0;
+    UINT64 lastPacketSentTimestamp = 0;
+
     CHK(pKvsRtpTransceiver != NULL, STATUS_NULL_ARG);
     pKvsPeerConnection = pKvsRtpTransceiver->pKvsPeerConnection;
     pPayloadArray = &(pKvsRtpTransceiver->sender.payloadArray);
-    ATOMIC_ADD(&pKvsRtpTransceiver->sender.outboundRtpStreamStats.totalEncodedBytesTarget, pFrame->size);
     if (MEDIA_STREAM_TRACK_KIND_VIDEO == pKvsRtpTransceiver->sender.track.kind) {
-        ATOMIC_INCREMENT(&pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded);
+        frames++;
         if (0 != (pFrame->flags & FRAME_FLAG_KEY_FRAME)) {
-            ATOMIC_INCREMENT(&pKvsRtpTransceiver->sender.outboundRtpStreamStats.keyFramesEncoded);
+            keyframes++;
         }
         if (pKvsRtpTransceiver->sender.lastKnownFrameCountTime == 0) {
             pKvsRtpTransceiver->sender.lastKnownFrameCountTime = now;
-            pKvsRtpTransceiver->sender.lastKnownFrameCount     = pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded;
+            pKvsRtpTransceiver->sender.lastKnownFrameCount = pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded;
         } else if (now - pKvsRtpTransceiver->sender.lastKnownFrameCountTime > HUNDREDS_OF_NANOS_IN_A_SECOND) {
-            UINT64 frames = pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded - pKvsRtpTransceiver->sender.lastKnownFrameCount;
-            UINT64 time   = now - pKvsRtpTransceiver->sender.lastKnownFrameCountTime;
-            DOUBLE fps    = (DOUBLE)(frames * HUNDREDS_OF_NANOS_IN_A_SECOND) / (DOUBLE) time;
-            ATOMIC_STORE(&pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesPerSecond, fps);
-            pKvsRtpTransceiver->sender.lastKnownFrameCountTime = now;
-            pKvsRtpTransceiver->sender.lastKnownFrameCount     = pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded;
+            UINT64 _frames = pKvsRtpTransceiver->sender.outboundRtpStreamStats.framesEncoded - pKvsRtpTransceiver->sender.lastKnownFrameCount;
```

Comment:
Perhaps name this something other than _frames and do the definition up at the beginning of the function with other params.